examples/templates/kubernetes/main.tf

```diff
@@ -137,5 +142,23 @@ resource "kubernetes_pod" "main" {
         read_only  = false
       }
     }
+
+    affinity {
+      pod_anti_affinity {
+        preferred_during_scheduling_ignored_during_execution {
+          weight = 1
+          pod_affinity_term {
+            topology_key = "kubernetes.io/hostname"
+            label_selector {
+              match_expressions {
+                key      = "app.kubernetes.io/name"
+                operator = "In"
+                values   = ["coder-workspace"]
+              }
+            }
+          }
+        }
+      }
```

Comment:
I'm confused, because wouldn't this fail for the defaults on most Kubernetes cluster that don't have the hostname of `coder-workspace`?

Reply:
It's matching pods that have `app.kubernetes.io/name: coder-workspace`, summing the amount partitioned by hostname. Our helm chart has something similar

Reply:
Ahh, I see. So even if you didn't deploy Coder with Kubernetes, this would still work? We should add a comment here.

Reply:
This is part of the [spec](https://kubernetes.io/docs/reference/labels-annotations-taints/#kubernetesiohostname), so I imagine any k8s-like apis would also implement this. I found some references to it in the openshift docs: https://docs.openshift.com/container-platform/4.8/nodes/scheduling/nodes-scheduler-node-selectors.html#nodes-scheduler-node-selectors-about_nodes-scheduler-node-selectors

Reply:
Oh I think I understand your question. This doesn't have anything to do with how Coder itself is deployed.